src/components/InfiniteLoader.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+
+import * as React from 'react';
+import { action, observable } from 'mobx';
+import { observer } from 'mobx-react';
+import { ActivityIndicator, Platform } from 'react-native';
+
+type Props = {|
+  fetchNextData: () => Promise<void>,
+|};
+
+export type InfiniteLoaderChildProps = {
+  loadingIndicator: React.Node,
+  onEndReached: () => Promise<void>,
+  onEndReachedThreshold: number,
+};
+
+@observer
+class InfiniteLoader extends React.Component<Props> {
+  @observable _loading = false;
+
+  @action
+  _setLoading = (loading: boolean) => {
+    this._loading = loading;
+  };
+
+  componentWillMount() {
+    this._fetchNextData();
+  }
+
+  _loadingIndicator = (): React.Node =>
+    this._loading ? <ActivityIndicator animating size="large" /> : null;
+
+  _fetchNextData = async (): Promise<void> => {
+    this._setLoading(true);
```

Comment:
What does using an action get you here? I would just set the property itself since it's an observable.

Reply:
in mobx.strict mode, we have to change any observable values only within actions..I thought that what you wanted at the first place?

Reply:
Ok, sounds good. I just wasn't sure about it.